spinoso-symbol/src/lib.rs

```diff
@@ -0,0 +1,370 @@
+pub struct SymbolOverflowError {
+    _private: (),
+}
+
+impl SymbolOverflowError {
+    /// The maximum identifier of a `Symbol`.
+    pub const MAX_CAPACITY: usize = u32::MAX as usize;
+
+    /// Construct a new, default `SymbolOverflowError`.
+    #[inline]
+    #[must_use]
+    pub const fn new() -> Self {
+        Self { _private: () }
+    }
+}
+
+impl From<TryFromIntError> for SymbolOverflowError {
+    #[inline]
+    fn from(err: TryFromIntError) -> Self {
+        let _ = err;
+        Self::new()
+    }
+}
+
+impl fmt::Display for SymbolOverflowError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_str("Symbol overflow")
+    }
+}
+
+#[cfg(feature = "std")]
+impl std::error::Error for SymbolOverflowError {}
+
+/// Identifier bound to an interned bytestring.
+///
+/// A `Symbol` allows retrieving a reference to the original interned
+/// bytestring.  Equivalent `Symbol`s will resolve to an identical bytestring.
+///
+/// `Symbol`s are based on a `u32` index. They are cheap to compare and cheap to
+/// copy.
+///
+/// `Symbol`s are not constrained to the interner which created them.
+#[derive(Debug, Clone, Copy, Hash, PartialEq, Eq, PartialOrd, Ord)]
+pub struct Symbol(u32);
+
+impl Symbol {
+    /// Construct a new `Symbol` from the given `u32`.
+    ///
+    /// `Symbol`s constructed manually may fail to resolve to an underlying
+    /// bytesstring.
+    ///
+    /// `Symbol`s are not constrained to the interner which created them.
+    /// No runtime checks ensure that the underlying interner is called with a
+    /// `Symbol` that the interner itself issued.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use spinoso_symbol::Symbol;
+    /// let sym = Symbol::new(263);
+    /// assert_eq!(sym.id(), 263);
+    /// ```
+    #[inline]
+    #[must_use]
+    pub const fn new(id: u32) -> Self {
+        Self(id)
+    }
+
+    /// Return the `u32` identifier from this `Symbol`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use spinoso_symbol::Symbol;
+    /// let sym = Symbol::new(263);
+    /// assert_eq!(sym.id(), 263);
+    /// assert_eq!(u32::from(sym), 263);
+    /// ```
+    #[inline]
+    #[must_use]
+    pub const fn id(self) -> u32 {
+        self.0
+    }
+
+    /// Returns whether the symbol is the empty byteslice `b""` in the
+    /// underlying interner.
+    ///
+    /// If there symbol does not exist in the underlying interner or there is an
+    /// error looking up the symbol in the underlying interner, `true` is
+    /// returned.
+    #[inline]
+    #[must_use]
+    #[cfg(feature = "artichoke")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "artichoke")))]
+    pub fn is_empty<T, U>(self, interner: &T) -> bool
+    where
+        T: Intern<Symbol = U>,
+        U: Copy + From<Symbol>,
+    {
+        if let Ok(Some(bytes)) = interner.lookup_symbol(self.into()) {
+            bytes.is_empty()
+        } else {
+            true
+        }
```

Comment:
This logic was straight up lifted from the implementation in `extn::core::symbol`.